.github/workflows/build_and_test.yml

```diff
@@ -8,28 +8,123 @@ on:
-    runs-on: ubuntu-latest
+  linux:
+    name: Linux
+    runs-on: ubuntu-20.04
 
     steps:
-      - name: Checkout repo
+      - name: Checkout
         uses: actions/checkout@v2.3.4
 
-      - name: Install Node.js
+      - name: Install Node
         uses: actions/setup-node@v2.2.0
         with:
           node-version: 16
           cache: npm
       
-      - name: Install dependencies
+      - name: Install NPM Dependencies
         run: npm ci
 
-      - name: Build web app
+      - name: Build Web App
         run: npx gulp build --platform=browser
-      
-      - name: Test web app
+
+      - name: Test Web App
         run: npm run action scripts/test
 
-      - name: Build Electron app
-        run: npm run action src/electron/build
+      - name: Build Electron App
+        run: npm run action src/electron/package_linux
+
+  windows:
+    name: Windows
+    runs-on: windows-2019
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Build Windows Client
+        run: npm run action src/electron/package_windows
+
+  mac:
+    name: MacOS
+    runs-on: macos-11
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Set XCode Version
+        run: sudo xcode-select -switch /Applications/Xcode_12.5.app
+
+      - name: Build OSX Client
+        run: npx gulp build --platform=osx
+
+  ios:
+    name: iOS
+    runs-on: macos-11
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Set XCode Version
+        run: sudo xcode-select -switch /Applications/Xcode_12.5.app
+
+      - name: Build iOS Client
+        run: npm run action apple/scripts/ios_package_remote
+
+  android:
+    name: Android
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Install Node
+        uses: actions/setup-node@v2.2.0
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Install NPM Dependencies
+        run: npm ci
+
+      - name: Install Java
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Setup Android
+        uses: android-actions/setup-android@v2
+
+      - name: Set Android SDK Version
+        run: |
+          sdkmanager "platforms;android-29"
+          sdkmanager --uninstall "build-tools;31.0.0"
```

Comment:
https://github.com/apache/cordova-android/issues/1288

this is... well.  😅  embarrassing? apparently `build-tools;31.0.0` is corrupted? How should I proceed here @fortuna? This works but feels... wrong. do I try to upgrade to `cordova-android@10.0.0`?